lib/auth/permissions.go

```diff
@@ -181,6 +181,38 @@ func GetCheckerForBuiltinRole(role teleport.Role) (services.AccessChecker, error
 				},
 			})
 	case teleport.RoleProxy:
+		if clusterConfig.GetSessionRecording() == services.RecordAtProxy {
```

Comment:
a comment here would be nice. what's going on? seems like a special hard-coded role is getting defined.